// @morev/utils/dist/index.js
import "./chunk-5DNDSZ5N.js";
import {
  promiseController
} from "./chunk-PRKUJMFN.js";
import {
  minTimePromise
} from "./chunk-WVNRJQVA.js";
import "./chunk-OVBUWFYF.js";
import {
  rangesInvert
} from "./chunk-HTSBC6SA.js";
import {
  rangeIncludes
} from "./chunk-SLQKMIUS.js";
import {
  rangesCrop
} from "./chunk-D3FEA4Q3.js";
import {
  rangesMerge
} from "./chunk-YAHJCUW4.js";
import {
  rangesSort
} from "./chunk-IITSDSVJ.js";
import "./chunk-TDPEW6GZ.js";
import "./chunk-URTX445H.js";
import "./chunk-MYDG7TVA.js";
import "./chunk-BO65XCYB.js";
import "./chunk-4NZEMUJZ.js";
import "./chunk-QQKCKQSZ.js";
import "./chunk-ZXPRZNUR.js";
import "./chunk-LNFQO5XW.js";
import "./chunk-USAQCG43.js";
import "./chunk-WRQLGYYR.js";
import "./chunk-B44LJQST.js";
import "./chunk-I2QBUDGO.js";
import "./chunk-YZAYOV42.js";
import "./chunk-HDFVHHLJ.js";
import "./chunk-IZGG7HHG.js";
import "./chunk-OGIYE2RH.js";
import "./chunk-NOALSN22.js";
import "./chunk-ON3223NI.js";
import "./chunk-O52LKZH7.js";
import "./chunk-OTICJJSN.js";
import "./chunk-TTZZQRNK.js";
import {
  queryStringify
} from "./chunk-7PSHRKIX.js";
import "./chunk-FADNZEVI.js";
import {
  getElement
} from "./chunk-HAKRSJIV.js";
import {
  getElementOffset
} from "./chunk-E73OSVHM.js";
import {
  getScrollLimit
} from "./chunk-AZKEOTOA.js";
import {
  getDocumentSize
} from "./chunk-JKOWNRAJ.js";
import {
  getScrollableAncestor
} from "./chunk-46E5IX4E.js";
import {
  getScrollbarWidth
} from "./chunk-J7NKYSCD.js";
import {
  getWindowScroll
} from "./chunk-4JXY2WW4.js";
import "./chunk-2LYOD43H.js";
import {
  isOSX
} from "./chunk-WCAQQIST.js";
import {
  isProd
} from "./chunk-R7DDKMRH.js";
import {
  isServer
} from "./chunk-TTSFIVKA.js";
import {
  isTest
} from "./chunk-PJOEEWLD.js";
import {
  isTouch
} from "./chunk-SEJGFAF3.js";
import {
  supportsPassive
} from "./chunk-MF6XUJNR.js";
import {
  isDev
} from "./chunk-7TAAL7TZ.js";
import {
  isFirefox
} from "./chunk-TVMLY5SN.js";
import {
  isInternetExplorer
} from "./chunk-32TFFOFK.js";
import {
  isLighthouse
} from "./chunk-GTUT74YY.js";
import {
  isClient
} from "./chunk-CRYBHWGN.js";
import {
  isMobile
} from "./chunk-L6JF55BK.js";
import {
  isMotionless
} from "./chunk-PYDMVANF.js";
import "./chunk-C4PFADZD.js";
import {
  verifyFileAccept
} from "./chunk-5YKDGE6O.js";
import {
  withProbability
} from "./chunk-ERLWIKXG.js";
import "./chunk-JLJPKFSA.js";
import {
  compose
} from "./chunk-UMXLHLRO.js";
import {
  debounce
} from "./chunk-V4YXQERH.js";
import {
  hash
} from "./chunk-57JKRZYH.js";
import {
  noop
} from "./chunk-MR7JPI4H.js";
import {
  safeJsonParse
} from "./chunk-TORW3K7T.js";
import {
  sleep
} from "./chunk-HROAT5GA.js";
import {
  throttle
} from "./chunk-72DZNZFR.js";
import {
  assert,
  configureAssert
} from "./chunk-JAZU2HBO.js";
import "./chunk-DVNKNPTJ.js";
import "./chunk-7EUPRUSV.js";
import {
  distance
} from "./chunk-UT2PMYRL.js";
import {
  lerp
} from "./chunk-R7SKROQY.js";
import "./chunk-PSAKS244.js";
import {
  parseResponseCookie
} from "./chunk-EYALVPWB.js";
import "./chunk-SUWZHXGV.js";
import {
  serializeCookie
} from "./chunk-23JLJNXY.js";
import "./chunk-OHMNL74K.js";
import {
  createFormatDate,
  formatDate,
  formatDateEn,
  formatDateRu
} from "./chunk-JAL6VPDP.js";
import {
  toDate
} from "./chunk-TGTLNHTH.js";
import {
  FORMAT_DATE_LOCALE_EN,
  FORMAT_DATE_LOCALE_RU
} from "./chunk-M5HBLC4T.js";
import "./chunk-ZH3ZSQ4E.js";
import {
  isIsoDate
} from "./chunk-F54PMRW7.js";
import {
  isLeapYear
} from "./chunk-NBFOTW2C.js";
import {
  parseDate
} from "./chunk-BYXCM36C.js";
import {
  randomDate
} from "./chunk-DHKWGA7L.js";
import {
  daysInMonth
} from "./chunk-OI2ZBQLP.js";
import "./chunk-Y6YZ3JAL.js";
import "./chunk-4WFMUP35.js";
import {
  precision
} from "./chunk-ZCWDLJA4.js";
import {
  randomFloat
} from "./chunk-EWXRMA6N.js";
import {
  clamp
} from "./chunk-EJGCYSTB.js";
import {
  formatBytes,
  formatBytesRu
} from "./chunk-V3M2VBX5.js";
import {
  numberFormat
} from "./chunk-3S3O2ZHY.js";
import "./chunk-5LECPUSN.js";
import {
  objectGet
} from "./chunk-7OI2NOV6.js";
import {
  objectKeysCase
} from "./chunk-4NUN7VQX.js";
import {
  objectToFormdata
} from "./chunk-F5GUYLBM.js";
import {
  omit
} from "./chunk-WFLW3WOP.js";
import {
  pick
} from "./chunk-ZXVPBFYX.js";
import {
  tsObject
} from "./chunk-K2EWJMCD.js";
import {
  deepClone,
  deepCloneStrict
} from "./chunk-FFWC26DF.js";
import {
  deepEqual,
  deepEqualCircular
} from "./chunk-5YCTAAGH.js";
import {
  formDataToObject
} from "./chunk-HLJ6FGAV.js";
import {
  toNumber
} from "./chunk-OCZQGT2X.js";
import "./chunk-OSBSRYIM.js";
import {
  randomString
} from "./chunk-SBX5PDGJ.js";
import {
  romanize
} from "./chunk-CGRZQPAE.js";
import {
  snakeCase
} from "./chunk-7SVUEZ2T.js";
import {
  stripIndent
} from "./chunk-KLZZ775T.js";
import {
  unquote
} from "./chunk-U7REZOM6.js";
import {
  escapeRegExp
} from "./chunk-4APQF3QV.js";
import {
  fileExtension
} from "./chunk-AMA5DX26.js";
import {
  formatSlashes
} from "./chunk-MLONGT6L.js";
import {
  kebabCase
} from "./chunk-KJRDAHBD.js";
import {
  padEnd
} from "./chunk-DENZN3B2.js";
import {
  padStart
} from "./chunk-WDLKPA7O.js";
import {
  pascalCase
} from "./chunk-LG4UJQF2.js";
import {
  quote
} from "./chunk-ABYGLOZM.js";
import {
  camelCase
} from "./chunk-T5PJGW6H.js";
import {
  capitalize
} from "./chunk-CJ426L5V.js";
import {
  createMergeObjects,
  mergeObjects
} from "./chunk-SYYF6SU6.js";
import "./chunk-CZZG4PNH.js";
import {
  asyncArray
} from "./chunk-XGD24OZP.js";
import "./chunk-GSJOLHOQ.js";
import {
  toArray
} from "./chunk-OAZLYBJA.js";
import {
  arrayRange
} from "./chunk-QSUEZT35.js";
import {
  arrayRemove,
  arrayRemoveMutable
} from "./chunk-EMZR7HH7.js";
import {
  arraySample
} from "./chunk-CMFCVGKH.js";
import {
  randomInteger
} from "./chunk-FDBXWY5V.js";
import {
  arrayOfLength
} from "./chunk-36MHHFDE.js";
import {
  arrayShuffle
} from "./chunk-IECQATAF.js";
import {
  arrayUnique
} from "./chunk-FER4ZC5Z.js";
import {
  arraysDifference
} from "./chunk-26GDFEXO.js";
import {
  arraysUnion
} from "./chunk-XXCXULIL.js";
import {
  arraysIntersection
} from "./chunk-4TKPKUM3.js";
import {
  arrayInsert,
  arrayInsertMutable
} from "./chunk-XYOXRCP6.js";
import "./chunk-BT72POBR.js";
import {
  isSymbol
} from "./chunk-TKM2FQTA.js";
import {
  isWeakMap
} from "./chunk-Y3T3BIRX.js";
import {
  isPlainObject
} from "./chunk-EWB5CQOX.js";
import {
  isPrimitive
} from "./chunk-XIKQVDFD.js";
import {
  isPromise
} from "./chunk-SPK7OES6.js";
import {
  isRegExp
} from "./chunk-5HZ32HEL.js";
import {
  isSessionStorageAvailable
} from "./chunk-2HW42MFA.js";
import {
  isHex
} from "./chunk-GEEDF44I.js";
import {
  isInteger
} from "./chunk-4W3EM56D.js";
import {
  isLocalStorageAvailable
} from "./chunk-MKT6NLMZ.js";
import {
  isMap
} from "./chunk-EXCPRRER.js";
import {
  isNode
} from "./chunk-M3TQVFXR.js";
import {
  isNullish
} from "./chunk-XDUT4IRC.js";
import {
  isUndefined
} from "./chunk-Z2T7SWRD.js";
import {
  isNull
} from "./chunk-TK4YRMRC.js";
import {
  isEmail
} from "./chunk-3SNQM74H.js";
import {
  isEmpty
} from "./chunk-76QAEAB7.js";
import {
  isObject
} from "./chunk-XUWL4VKZ.js";
import {
  isIterable
} from "./chunk-HJKYC4E4.js";
import {
  isError
} from "./chunk-WE7QC4OC.js";
import {
  isFile
} from "./chunk-TZZM7RNI.js";
import {
  isFloat
} from "./chunk-MV2MTLNA.js";
import {
  isNumeric
} from "./chunk-OBG63REY.js";
import {
  isFormData
} from "./chunk-ON2BNX4R.js";
import {
  isFunction
} from "./chunk-CQJCSTBU.js";
import {
  isArray
} from "./chunk-UDVSPJHY.js";
import {
  isBlob
} from "./chunk-UDW3FTZT.js";
import {
  isBoolean
} from "./chunk-42JGV6MK.js";
import {
  isDate
} from "./chunk-4GTX7FBA.js";
import {
  isElement
} from "./chunk-FMEA3XAZ.js";
import {
  isString
} from "./chunk-RUB2ELNH.js";
export {
  FORMAT_DATE_LOCALE_EN,
  FORMAT_DATE_LOCALE_RU,
  arrayInsert,
  arrayInsertMutable,
  arrayOfLength,
  arrayRange,
  arrayRemove,
  arrayRemoveMutable,
  arraySample,
  arrayShuffle,
  arrayUnique,
  arraysDifference,
  arraysIntersection,
  arraysUnion,
  assert,
  asyncArray,
  camelCase,
  capitalize,
  clamp,
  compose,
  configureAssert,
  createFormatDate,
  createMergeObjects,
  daysInMonth,
  debounce,
  deepClone,
  deepCloneStrict,
  deepEqual,
  deepEqualCircular,
  distance,
  escapeRegExp,
  fileExtension,
  formDataToObject,
  formatBytes,
  formatBytesRu,
  formatDate,
  formatDateEn,
  formatDateRu,
  formatSlashes,
  getDocumentSize,
  getElement,
  getElementOffset,
  getScrollLimit,
  getScrollableAncestor,
  getScrollbarWidth,
  getWindowScroll,
  hash,
  isArray,
  isBlob,
  isBoolean,
  isClient,
  isDate,
  isDev,
  isElement,
  isEmail,
  isEmpty,
  isError,
  isFile,
  isFirefox,
  isFloat,
  isFormData,
  isFunction,
  isHex,
  isInteger,
  isInternetExplorer,
  isIsoDate,
  isIterable,
  isLeapYear,
  isLighthouse,
  isLocalStorageAvailable,
  isMap,
  isMobile,
  isMotionless,
  isNode,
  isNull,
  isNullish,
  isNumeric,
  isOSX,
  isObject,
  isPlainObject,
  isPrimitive,
  isProd,
  isPromise,
  isRegExp,
  isServer,
  isSessionStorageAvailable,
  isString,
  isSymbol,
  isTest,
  isTouch,
  isUndefined,
  isWeakMap,
  kebabCase,
  lerp,
  mergeObjects,
  minTimePromise,
  noop,
  numberFormat,
  objectGet,
  objectKeysCase,
  objectToFormdata,
  omit,
  padEnd,
  padStart,
  parseDate,
  parseResponseCookie,
  pascalCase,
  pick,
  precision,
  promiseController,
  queryStringify,
  quote,
  randomDate,
  randomFloat,
  randomInteger,
  randomString,
  rangeIncludes,
  rangesCrop,
  rangesInvert,
  rangesMerge,
  rangesSort,
  romanize,
  safeJsonParse,
  serializeCookie,
  sleep,
  snakeCase,
  stripIndent,
  supportsPassive,
  throttle,
  toArray,
  toDate,
  toNumber,
  tsObject,
  unquote,
  verifyFileAccept,
  withProbability
};
